### ISSUE TITLE
prefer_self_in_static_references has false positive on default parameters

### ISSUE THREAD

**Issue by dalemyers** [realm/SwiftLint#3876]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Describe the bug

This bug is similar to https://github.com/realm/SwiftLint/issues/3775 

Default parameters are falsely flagged by this rule. You cannot use `Self` in this location. 

```swift
public final class Foo {
  static var foo = Foo()

  func function(f: Foo = Foo.foo) {}
}
```

SwiftLint version: 0.46.3

**Comment by dalemyers** [comment]
I'm not sure what happened but I suspect I had two versions installed on my machine. One which was running via scripts but was older and still had this issue, and the other which was 0.46.3. I no longer repro this.
